python/dgllife/data/pdbbind.py

```diff
@@ -73,34 +78,132 @@ class PDBBind(object):
+                for target_set in ['train', 'valid', 'test']]
+            self.agg_structure_split = [list(merged_df.loc[merged_df['structure']==target_set, 'PDB_code'].index) 
+                for target_set in ['train', 'valid', 'test']]
 
-        self._url = 'dataset/pdbbind_v2015.tar.gz'
+    def _read_data_files(self, pdb_version, subset, load_binding_pocket, remove_coreset_from_refinedset, local_path):
+        """Download and extract pdbbind data files specified by the version"""
         root_dir_path = get_download_dir()
-        data_path = root_dir_path + '/pdbbind_v2015.tar.gz'
-        extracted_data_path = root_dir_path + '/pdbbind_v2015'
-        download(_get_dgl_url(self._url), path=data_path, overwrite=False)
-        extract_archive(data_path, extracted_data_path)
-
-        if subset == 'core':
-            index_label_file = extracted_data_path + '/v2015/INDEX_core_data.2013'
-        elif subset == 'refined':
-            index_label_file = extracted_data_path + '/v2015/INDEX_refined_data.2015'
-        else:
-            raise ValueError(
-                'Expect the subset_choice to be either '
+        if local_path:
+            if local_path[-1]!='/':
+                local_path += '/'
+            index_label_file = glob.glob(local_path + '*' + subset + '*data*')[0]
+        elif pdb_version == 'v2015':
+            self._url = 'dataset/pdbbind_v2015.tar.gz'
+            data_path = root_dir_path + '/pdbbind_v2015.tar.gz'
+            extracted_data_path = root_dir_path + '/pdbbind_v2015'
+            download(_get_dgl_url(self._url), path=data_path, overwrite=False)
+            extract_archive(data_path, extracted_data_path)
+
+            if subset == 'core':
+                index_label_file = extracted_data_path + '/v2015/INDEX_core_data.2013'
+            elif subset == 'refined':
+                index_label_file = extracted_data_path + '/v2015/INDEX_refined_data.2015'
+            else:
+                raise ValueError(
+                    'Expect the subset_choice to be either '
                 'core or refined, got {}'.format(subset))
+        elif pdb_version == 'v2007':
+            # download from pddbind official website
+        #     download_url = 'http://www.pdbbind.org.cn/download/pdbbind_v2007.tar.gz' 
+        #     data_path = root_dir_path + '/pdbbind_v2007.tar.gz'
+        #     extracted_data_path = root_dir_path + '/pdbbind_v2007'
+        #     download(download_url, path=data_path, overwrite=False)
+        #     extract_archive(data_path, extracted_data_path)
+            self._url = 'dataset/pdbbind_v2007.tar.gz'
+            data_path = root_dir_path + '/pdbbind_v2007.tar.gz'
+            extracted_data_path = root_dir_path + '/pdbbind_v2007'
+            download(_get_dgl_url(self._url), path=data_path, overwrite=False)
+            extract_archive(data_path, extracted_data_path, overwrite=False)
+            extracted_data_path += '/home/ubuntu' # extra layer 
```

Comment:
This does not appear correct as it will not work on other OS like Windows or Mac OS.

Reply:
It's the extra layer that comes from downloading from DGL database

Reply:
Can you give me a pointer?

Reply:
https://data.dgl.ai/dataset/pdbbind_v2007.tar.gz